torch/autograd/graph.py

```diff
@@ -536,11 +536,33 @@ def wrapped_fn(grad: torch.Tensor):
 
 
 def _get_tid(t) -> Tuple[int, int, int]:
-    return (id(t), t.data_ptr(), t._version)
+    # FIXME: This is almost definitely a bug.
```

Comment:
😮 does the autograd engine use data_ptr somewhere

Reply:
Hmm not that I know of. For context though, the code here is a utility independent of the autograd engine that uses TorchDispatchMode to dynamically insert clones whenever user does in-place on a tensor that has been saved for backward. We're using data_ptr as a way to track storage aliasing, so if a particular tensor is in-placed, any tensor saved for backward aliasing that tensor can also be cloned.

Reply:
We have storage pyobject preservation now @ezyang told me.
We can use a that as a unique id instead of data_ptr!